sys-cluster/openmpi/openmpi-1.10.2-r1.ebuild

```diff
@@ -0,0 +1,201 @@
+	openmpi_ofed_features_rdmacm
+	openmpi_ofed_features_dynamic-sl
+	openmpi_ofed_features_failover"
+
+DESCRIPTION="A high-performance message passing library (MPI)"
+HOMEPAGE="http://www.open-mpi.org"
+SRC_URI="http://www.open-mpi.org/software/ompi/v$(get_version_component_range 1-2)/downloads/${MY_P}.tar.bz2"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS=""
+IUSE="cma cuda +cxx elibc_FreeBSD fortran heterogeneous ipv6 java mpi-threads numa romio threads vt
+	${IUSE_OPENMPI_FABRICS} ${IUSE_OPENMPI_RM} ${IUSE_OPENMPI_OFED_FEATURES}"
+
+REQUIRED_USE="openmpi_rm_slurm? ( !openmpi_rm_pbs )
+	openmpi_rm_pbs? ( !openmpi_rm_slurm )
+	openmpi_fabrics_psm? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_control-hdr-padding? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_connectx-xrc? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_udcm? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_rdmacm? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_dynamic-sl? ( openmpi_fabrics_ofed )
+	openmpi_ofed_features_failover? ( openmpi_fabrics_ofed )"
+
+MPI_UNCLASSED_DEP_STR="
+	vt? (
+		!dev-libs/libotf
+		!app-text/lcdf-typetools
+	)"
+
+# dev-util/nvidia-cuda-toolkit is always multilib
+CDEPEND="
+	!sys-cluster/mpich
+	!sys-cluster/mpich2
+	!sys-cluster/mpiexec
+	>=dev-libs/libevent-2.0.21[${MULTILIB_USEDEP}]
+	dev-libs/libltdl:0[${MULTILIB_USEDEP}]
+	>=sys-apps/hwloc-1.10.0-r2[${MULTILIB_USEDEP},numa?]
+	>=sys-libs/zlib-1.2.8-r1[${MULTILIB_USEDEP}]
+	cuda? ( >=dev-util/nvidia-cuda-toolkit-6.5.19-r1 )
+	elibc_FreeBSD? ( dev-libs/libexecinfo )
+	openmpi_fabrics_ofed? ( sys-fabric/ofed:* )
+	openmpi_fabrics_knem? ( sys-cluster/knem )
+	openmpi_fabrics_psm? ( sys-fabric/infinipath-psm:* )
+	openmpi_rm_pbs? ( sys-cluster/torque )
+	openmpi_rm_slurm? ( sys-cluster/slurm )
+	openmpi_ofed_features_rdmacm? ( sys-fabric/librdmacm:* )"
+
+RDEPEND="${CDEPEND}
+	java? ( >=virtual/jre-1.6 )"
+
+DEPEND="${CDEPEND}
+	java? ( >=virtual/jdk-1.6 )"
+
+MULTILIB_WRAPPED_HEADERS=(
+	/usr/include/mpi.h
+)
+
+pkg_setup() {
+	fortran-2_pkg_setup
+	java-pkg-opt-2_pkg_setup
+
+	if use mpi-threads; then
+		echo
+		ewarn "WARNING: use of MPI_THREAD_MULTIPLE is still disabled by"
+		ewarn "default and officially unsupported by upstream."
+		ewarn "You may stop now and set USE=-mpi-threads"
+		echo
+	fi
+
+	echo
+	elog "OpenMPI has an overwhelming count of configuration options."
+	elog "Don't forget the EXTRA_ECONF environment variable can let you"
+	elog "specify configure options if you find them necessary."
+	echo
+}
+
+src_prepare() {
+	# Necessary for scalibility, see
+	# http://www.open-mpi.org/community/lists/users/2008/09/6514.php
+	if use threads; then
+		echo 'oob_tcp_listen_mode = listen_thread' \
+			>> opal/etc/openmpi-mca-params.conf
+	fi
+}
+
+multilib_src_configure() {
+	local myconf=(
+		--sysconfdir="${EPREFIX}/etc/${PN}"
+		--enable-pretty-print-stacktrace
+		--enable-orterun-prefix-by-default
+		--with-hwloc="${EPREFIX}/usr"
+		--with-libltdl="${EPREFIX}/usr"
+		)
```

Comment:
attach all of this stuff directly to `econf`, seeing the next block is not needed anymore